### DISCUSSION TITLE
Brew uninstalled more than I expected

### DISCUSSION THREAD

**Discussion by cybenv** [Homebrew/discussions#5432] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.3-71-g633b46a-dirty
ORIGIN: https://github.com/Homebrew/brew
HEAD: 633b46a7c293cb6c5a974ea060d8ec37ae2a9bb0
Last commit: 7 hours ago
Core tap JSON: 08 Jun 00:10 UTC
Core cask tap JSON: 08 Jun 00:10 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.2 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.2/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.45.2 => /usr/local/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 13.6.7-x86_64
CLT: 15.1.0.0.1.1700200546
Xcode: 15.1
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Hey everyone,

I used the `brew uninstall --cask chainner` command to remove an application and was surprised to see dependencies like gcc, numpy, openblas, etc., automatically removed, even though I didn't use **brew autoremove**. Usually, the dependencies remain unless I decide to remove them myself. Can you help me figure out if this is normal?

N.B. I didn't set any special configurations for brew that could automatically remove dependencies (HOMEBREW_CASK_OPTS is empty). Everything was default.

Homebrew version: 4.3.3-71-g633b46a-dirty
MacOS 13.6.7

Thank you!

**Comment by jacktose** [comment, marked as answer]
https://github.com/Homebrew/brew/pull/17261 enabled this function by default. `export HOMEBREW_NO_AUTOREMOVE=1` to disable it.
(Note that there was some debate about this choice: https://github.com/Homebrew/brew/pull/17364, https://github.com/Homebrew/brew/pull/17369.)
